### PR TITLE
Add missing flags for hip::device target on hip-clang

### PR DESCRIPTION
**Pull request by pfultz2**
This adds the missing compilation flags to `hip::device` so it can compile with hip-clang compiler.

### CODE REVIEW

**Comment by daniellowell** [comment]
What is the status of this PR?

File: hip-config-clang.cmake.in [hunk 1]
```diff
@@ -50,9 +50,50 @@ set_and_check( hip_BIN_INSTALL_DIR "@PACKAGE_BIN_INSTALL_DIR@" )
...
+
 find_dependency(amd_comgr)
+find_dependency(AMDDeviceLibs)
+set(AMDGPU_TARGETS "gfx900;gfx906" CACHE STRING "AMD GPU targets to compile for")
+set(GPU_TARGETS "${AMDGPU_TARGETS}" CACHE STRING "GPU targets to compile for")
+
 include( "${CMAKE_CURRENT_LIST_DIR}/hip-targets.cmake" )
 
+set_property(TARGET hip::host APPEND PROPERTY 
+  INTERFACE_COMPILE_DEFINITIONS __HIP__=1
```
[the remark this answers is not in the document]
Comment (pfultz2) [reply hip-config-clang.cmake.in]: Fixed.

**Comment by mangupta** [comment]
@pfultz2 Is this still required? If so, @mhbliao can you please review it and approve the changes if it is ready to merge?

**Comment by pfultz2** [comment]
It is required for clang tidy to work with hip-clang. However, if the flags have changed then this needs to be update. I dont have access to hip-clang to test this yet.

**Comment by daniellowell** [comment]
@mangupta Can we have this move forward, or adjusted to move forward. We need this for static analysis in our CI. This is important to develop stable code during the transition to hip-clang

cc: @dagamayank

File: hip-config-clang.cmake.in [hunk 2]
```diff
@@ -50,9 +50,46 @@ set_and_check( hip_BIN_INSTALL_DIR "@PACKAGE_BIN_INSTALL_DIR@" )
...
+set_property(TARGET hip::device APPEND PROPERTY 
+  INTERFACE_COMPILE_OPTIONS -x hip --hip-device-lib-path=${AMD_DEVICE_LIBS_PREFIX}/lib
+)
+
+set_property(TARGET hip::device APPEND PROPERTY 
+  INTERFACE_LINK_LIBRARIES --hip-device-lib-path=${AMD_DEVICE_LIBS_PREFIX}/lib --hip-link
+)
+
+set_property(TARGET hip::device APPEND PROPERTY 
+  INTERFACE_INCLUDE_DIRECTORIES "${HIP_CLANG_INCLUDE_PATH}"
```
Comment (yxsamliu) [inline hip-config-clang.cmake.in]: After https://github.com/ROCm-Developer-Tools/HIP/pull/1917 is merged, you may need to include "${HIP_CLANG_INCLUDE_PATH}/.."

File: hip-config-clang.cmake.in [hunk 3]
```diff
@@ -50,9 +50,46 @@ set_and_check( hip_BIN_INSTALL_DIR "@PACKAGE_BIN_INSTALL_DIR@" )
...
+)
+
+set_property(TARGET hip::device APPEND PROPERTY 
+  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${HIP_CLANG_INCLUDE_PATH}"
```
Comment (yxsamliu) [inline hip-config-clang.cmake.in]: After https://github.com/ROCm-Developer-Tools/HIP/pull/1917 is merged, you may need to include "${HIP_CLANG_INCLUDE_PATH}/.."

**Pull request merged**